iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/executable/ExecutableManager.java

```diff
@@ -227,6 +228,8 @@ protected void saveToDir(ByteBuffer byteBuffer, String destination) throws IOExc
     try {
       Path path = Paths.get(destination);
       Files.deleteIfExists(path);
+      // Ensure the target directory exists.
+      SystemFileFactory.INSTANCE.makeDirIfNecessary(destination);
```

Comment:
The destination String is not a Dir path, it is a file path I believe.

Reply:
Yeah... I've noticed that and implemented fixes and am currently trying my changes on mac/linux/win each for x86_64 and aarch64... Just doing this in a fork in order to not spam the making list.